java/code/src/com/suse/manager/webui/controllers/ProductsController.java

```diff
@@ -319,6 +318,10 @@ public static String data(Request request, Response response, User user) {
                                             s.getStatus(),
                                             s.getChannels().stream().map(c ->
                                                     new ChannelJson(
+                                                            Opt.fold(Optional.ofNullable(
+                                                                            channelByLabel.get(c.getLabel())),
+                                                                    () -> -1L,
+                                                                    x -> x.getId()),
```

Comment:
Nitpick: (this is **totally** matter of taste, so disregard it, if you don't like it.)

I found an alternative syntax to be more readable:

```java
Optional.ofNullable(channelByLabel.get(c.getLabel()))
    .map(chan -> chan.getId())
    .orElse(-1L)
```

Reply:
I am always undecided which is better actually.
`Opt.fold` is more complex but `Optional.ofNullable` uses `.map()` and I don't like to map when I know it is just a single object that I have to work with, it is less intuitive in such a case. So either one way or the other in this case has pros and cons to me. And if you say the second one is clearer, I'm fine with it and I trust you more in the Java aspect. Changing accordingly :+1: